app/src/ui/changes/oversized-files-warning.tsx

```diff
@@ -0,0 +1,91 @@
+import * as React from 'react'
+import { Button } from '../lib/button'
+import { ButtonGroup } from '../lib/button-group'
+import { Dialog, DialogContent, DialogFooter } from '../dialog'
+import { LinkButton } from '../lib/link-button'
+import { Monospaced } from '../lib/monospaced'
+import { PathText } from '../lib/path-text'
+import { Dispatcher } from '../../lib/dispatcher'
+import { Repository } from '../../models/repository'
+import { ITrailer } from '../../lib/git/interpret-trailers'
+
+const GitLFSWebsiteURL =
+  'https://help.github.com/articles/versioning-large-files/'
+
+interface IOversizedFilesProps {
+  readonly oversizedFiles: ReadonlyArray<string>
+  readonly onDismissed: () => void
+  readonly dispatcher: Dispatcher
+  readonly commitSummary: string
+  readonly commitDescription: string | null
+  readonly repository: Repository
+  readonly trailers?: ReadonlyArray<ITrailer>
+}
+
+/** A dialog to display a list of files that are too large to commit. */
+export class OversizedFiles extends React.Component<IOversizedFilesProps> {
+  public constructor(props: IOversizedFilesProps) {
+    super(props)
+  }
+
+  public render() {
+    return (
+      <Dialog
+        id="oversized-files"
+        title={__DARWIN__ ? 'Files Too Large' : 'Files too large'}
+        onDismissed={this.props.onDismissed}
+        type="warning"
+      >
+        <DialogContent>
+          <p>
+            The following files are over 100MB.{' '}
+            <strong>
+              If you commit these files, you will no longer be able to push this
+              repository to GitHub.com.
+            </strong>
+          </p>
+          {this.renderFileList()}
+          <p className="recommendation">
+            We recommend you remove these files before committing or use{' '}
```

Comment:
> I think this message could be a bit friendlier.

I agree and think this recommendation would work great. I have gone ahead and made these changes and am happy to update this further if there's anything else that looks like it can be improved.